CMakeLists.txt

```diff
@@ -486,9 +486,27 @@ install_export_targets()
 # Installation and packaging:
 #------------------------------------------------------------------------------
 
-
-# packaging routines:
 if(NOT "${PACKAGE_TYPE}" STREQUAL "none")
+
+  # Set packaging type
+  if((DEFINED ENV{PACK_ZIP}) AND (NOT "$ENV{PACK_ZIP}" STREQUAL ""))
+    set(CPACK_GENERATOR "ZIP")
+  else()
+    # Set CPACK_GENERATOR based on PACKAGE_TYPE
+    if("${PACKAGE_TYPE}" STREQUAL "dmg")
+      set(CPACK_GENERATOR "DragNDrop")
+    elseif("${PACKAGE_TYPE}" STREQUAL "pkg")
+      set(CPACK_GENERATOR "productbuild")
+    elseif("${PACKAGE_TYPE}" STREQUAL "rpm")
+      set(CPACK_GENERATOR "RPM")
+    elseif("${PACKAGE_TYPE}" STREQUAL "deb")
+      set(CPACK_GENERATOR "DEB")
+    elseif("${PACKAGE_TYPE}" STREQUAL "nsis")
+      set(CPACK_GENERATOR "NSIS")
+    endif()
```

Comment:
Does it make sense to add a default case, and throw an error?

Reply:
And in the case you wait for the restructure. Maybe warn if someone tries to set "zip" in the usual way via package_type?
"Warning please use it like this: .."

Reply:
I added default cases to throw errors and added warnings.